From 47e16e5bcff38e7cde014b3c78f3f0b4be69739f Mon Sep 17 00:00:00 2001
From: "Gary D. Gregory" <ggregory@apache.org>
Date: Wed, 22 Feb 2012 19:22:57 +0000
Subject: [PATCH] [IO-302] ArrayIndexOutOfBoundsException in BOMInputStream
 when reading a file without BOM multiple times. Refactor tests to use both
 BOM and no-BOM test files.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/io/trunk@1292460 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/io/input/BOMInputStreamTest.java  | 24 ++++++++++++-------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/src/test/java/org/apache/commons/io/input/BOMInputStreamTest.java b/src/test/java/org/apache/commons/io/input/BOMInputStreamTest.java
index af1c0f4e53b..99cba275a69 100644
--- a/src/test/java/org/apache/commons/io/input/BOMInputStreamTest.java
+++ b/src/test/java/org/apache/commons/io/input/BOMInputStreamTest.java
@@ -362,14 +362,22 @@ public void testNoBoms() throws Exception {
     }
     
     public void testReadTwiceWithoutBOM() throws Exception {
-    	InputStream inputStream = this.getClass().getResourceAsStream("/org/apache/commons/io/testfileNoBOM.xml");
-    	Assert.assertNotNull(inputStream);
-    	BOMInputStream bomInputStream = new BOMInputStream(inputStream);
-    	bomInputStream.mark(1000000);
-
-    	this.readFile(bomInputStream);
-    	bomInputStream.reset();
-    	this.readFile(bomInputStream);
+        this.readBOMInputStreamTwice("/org/apache/commons/io/testfileNoBOM.xml");
+    }
+
+    public void testReadTwiceWithBOM() throws Exception {
+        this.readBOMInputStreamTwice("/org/apache/commons/io/testfileBOM.xml");
+    }
+
+    public void readBOMInputStreamTwice(String resource) throws Exception {
+        InputStream inputStream = this.getClass().getResourceAsStream(resource);
+        Assert.assertNotNull(inputStream);
+        BOMInputStream bomInputStream = new BOMInputStream(inputStream);
+        bomInputStream.mark(1000000);
+
+        this.readFile(bomInputStream);
+        bomInputStream.reset();
+        this.readFile(bomInputStream);
     }
 
     private void readFile(BOMInputStream bomInputStream) throws Exception {
